In [4]:
import numpy as np

## Задача 1

### Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите M(X), D(X).

В нашем случае Х - число выпадения правильных ответов

n - число испытаний. В нашем случае их 5

p - вероятность появления события. В нашем случае она равна 1/4 для каждого испытания

Соответственно, q = 1 - 1/4 = 3/4

In [5]:
# 𝑀(𝑋)=𝑛𝑝

m = 5 * 1/4
print(m)

1.25


In [6]:
# 𝐷(𝑋)=𝑛𝑝𝑞

d = 5 * 1/4 * 3/4
print(d)

0.9375


Закон распределения числа правильных ответов при простом угадывании будет выглядеть следующим образом:

Вводные данные все те же - n=5, p=1/4, q=3/4

In [5]:
def bernulli(k): 
    P = (np.math.factorial(5) / (np.math.factorial(k) * np.math.factorial(5 - k))) * (1/4)**k * (3/4)**(5-k)
    return P

In [6]:
bernulli(0)

0.2373046875

In [7]:
bernulli(1)

0.3955078125

In [8]:
bernulli(2)

0.263671875

In [9]:
bernulli(3)

0.087890625

In [10]:
bernulli(4)

0.0146484375

In [11]:
bernulli(5)

0.0009765625

Итак, закон распределения такой случайной величины:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = x)$</th>
        <td>0.237</td>
        <td>0.396</td>
        <td>0.234</td>
        <td>0.088</td>
        <td>0.015</td>
        <td>0.001</td>
    </tr>
</table>

## Задача 2

### Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

p - вероятность попадания в цель при одном выстреле, в нашем случае 0.01

n - количество выстрелов, в нашем случае 200

k - количество вероятных попаданий в цель, в нашем случае это диапазон 5 =< k <= 10

Поскольку вероятность попадания очень мала, а количество выстрелов относительно велико, то вероятность я буду искать по формуле Пуассона

In [20]:
def poisson_proba(k, lambda_):
    
    return (lambda_ ** k) * (np.exp(-lambda_)) / np.math.factorial(k)

In [21]:
lambda_ = 200 * 0.01
lambda_

2.0

Поскольку у нас имеется диапазон верятных попаданий в цель, я использую формулу с суммой вероятностей, которую мы рассмотрели на уроке:

In [22]:
sum(poisson_proba(k=i, lambda_=lambda_) for i in range(5, 11))

0.052644709119342686

Таким образом, искомая вероятность равна 0.053

## Задача 3

### В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?

#### Подсказка. Как посчитать P(X >= k) для величины X, имеющей распределение Пуассона? Здесь предстоит немножко покодить.

p - вероятность попадания в цель при одном выстреле, в нашем случае 0.01

k - количество вероятных попаданий в цель, в нашем случае это диапазон k >= 10 (хотя бы 10 раз - это значит 10 и более)

Pk - вероятность в нашем случае равна 0.9

Поскольку диапазон задан от 10 и выше, то я выражу его как 1 - остальные вероятности ( 0 < k < 10).

Дальше я не представляю как действовать, кроме как перебрать разное количество испытаний и отфильтровать те количества испытаний, в результате которых достигается необходимый результат (вероятность от 0.9 до 0.905, т.к. 0.905 уже округлится как 0.91)

In [165]:
s = []
for l in range (5000):
    lambda_ = l * 0.01
    result = 1 - sum(poisson_proba(k=i, lambda_=lambda_) for i in range(10))
    if 0.9 <= result <= 0.905:
        s.append(l)
print(s)

[1421, 1422, 1423, 1424, 1425, 1426, 1427, 1428, 1429, 1430, 1431, 1432]


Для уверенности проверю первое и последнее значение данного списка 

In [166]:
lambda_ = 1421 * 0.01
lambda_

14.21

In [167]:
1 - sum(poisson_proba(k=i, lambda_=lambda_) for i in range(10))

0.9001760722984115

In [168]:
lambda_ = 1432 * 0.01
lambda_

14.32

In [169]:
1 - sum(poisson_proba(k=i, lambda_=lambda_) for i in range(10))

0.9049063320941403